docs/source/app-dev/bindings-java/codegen.rst

```diff
@@ -380,3 +383,96 @@ The Java code generated for this variant is:
     public Published(Long year, String publisher) { /* ... */ }
     public Record toValue() { /* ... */ }
   }
+
+Parameterized types
+~~~~~~~~~~~~~~~~~~~
+
```

Comment:
I think we should maybe actually mention here that this section is only for completeness, and we don't expect users to make use of the `fromValue` and `toValue` methods, because they would typically come from a template that doesn't have any unbound type parameters.
@bethaitman-da: do you think this makes sense from a "user education" point of view?

Reply:
This sounds pretty important - if there are caveats that mean you don't have to read this, they should definitely go up front!